### PR TITLE
Commit Drive Upload Locations into Repo

### PR DESCRIPTION
**Pull request by as2388**
Review #32 first.

Commits the Drive upload locations to the repository by adding them to the configuration json.

Also:
 - Uploads TracedData to Drive
 - Requires the pipeline configuration json to be passed in to the run scripts explicitly

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -21,6 +13,12 @@ RUN if [ "$INSTALL_CPU_PROFILER" = "true" ]; then \
         rm -rf /pyflame; \
     fi
 
+# Make a directory for private credentials files
+RUN mkdir /auth
```
Comment (lukechurch) [inline Dockerfile]: I thought we'd changed this before? Maybe I'm thinking of a different place?

File: pipeline.py [hunk 2]
```diff
@@ -178,28 +183,48 @@
...
+        parsed_rapid_pro_token_file_url = urlparse(pipeline_configuration.drive_upload.drive_credentials_file_url)
+        bucket_name = parsed_rapid_pro_token_file_url.netloc
+        blob_name = parsed_rapid_pro_token_file_url.path.lstrip("/")
 
-        csv_by_individual_drive_dir = os.path.dirname(csv_by_individual_drive_path)
-        csv_by_individual_drive_file_name = os.path.basename(csv_by_individual_drive_path)
-        drive_client_wrapper.update_or_create(csv_by_individual_output_path, csv_by_individual_drive_dir,
-                                              target_file_name=csv_by_individual_drive_file_name,
-                                              target_folder_is_shared_with_me=True)
+        print(f"Downloading Drive service account credentials from file '{blob_name}' in bucket '{bucket_name}'...")
```
Comment (lukechurch) [inline pipeline.py]: I think this should be moved earlier. If the service account credentials fail, we shouldn't run the whole of the pipeline.

I agree the upload should happen last, but the prerequisites should be checked at the beginning of the pipeline.

**Review by lukechurch — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (as2388) [reply Dockerfile]: Yes, that just hadn't been merged when you reviewed this

File: pipeline.py (hunk 2 above)
Comment (as2388) [reply pipeline.py]: Moved

**Pull request merged**